### ISSUE TITLE
Multiple spaces in translated values prevent ordinals from working

### ISSUE THREAD

**Issue by burkmarr** [burkmarr/tombiovis#25]
From email from Leslie Reid dated 25/07/2018: Hi, Rich,

I’m now framing out my fourth key, and the software continues to work great. I did run across another unexpected behavior that appears to have resulted from a quirk in my labeling strategy; I’ve attached my “TestResult” spreadsheet that illustrates the problem and the solution I found. The problem was that I couldn’t get the “Mass1” character to provide results as an ordinal variable—it worked just fine as a text-multi or text-single, and other characters in the original (Russula) spreadsheet worked just fine as ordinals. If you turn this spreadsheet into a key, you’ll find that the first and third characters work fine, while the second registers nothing. The fourth reflects the solution.

I tried a number of things to isolate the problem, which turned out to be the way I’d translated the CharacterState in the “CharacterStateTranslation” field. When I put the input code in parentheses (see lines 36-45 in Values) instead of leading off with it (lines 14-23), everything worked fine.

And at this point I have to admit that I thoroughly enjoy working on this kind of puzzle!

**Comment by burkmarr** [comment]
My reply to Leslie: Well I’m kind of amazed by that – what a strange bug! I was able to get Mass3 to work as an ordinal by editing your original translated values to replace multiple spaces with a single space. So there’s some sort of bug around multiple spaces. I will log this as a bug and fix it for the next release. Thanks very much Leslie!

**Comment by burkmarr** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Multiple spaces in translated values prevent ordinals from working.](https://trello.com/c/owGltSHs/183-multiple-spaces-in-translated-values-prevent-ordinals-from-working)

**Comment by burkmarr** [comment]
Upon investigation I found that this was related to #8. The solution I implemented for that issue only took double spaces out of translated values where they were used to replace the values from the taxon tab. It left the translated values themselves on the values tab with the spaces, but these were then used when scoring ordinal values and therefore not matching. So I changed dealing with the double spaces so that they are removed from the translated values in the values tab itself right at the head of post-processing in tbv.f.loadcomplete.
